include/clap/ext/draft/scratch-memory.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "../../plugin.h"
+
+// This extension lets the plugin request "scratch" memory
+// from the host. Scratch memory can be accessed during the
+// `process()` callback, but is not persistent between callbacks.
+
+static CLAP_CONSTEXPR const char CLAP_EXT_SCRATCH_MEMORY[] = "clap.scratch-memory/1";
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct clap_host_scratch_memory {
+   // Asks the host for certain amount of scratch memory.
+   // If the host is unable to provide the memory, it should
+   // return "false".
+   //
+   // Note that any memory the host allocates to satisfy
+   // the requested scratch size can be de-allocated
+   // when the plugin is de-activated.
+   //
+   // [main-thread & being-activated]
+   bool(CLAP_ABI *request_size)(const clap_host_t *host, size_t scratch_size_bytes);
```

Comment:
We have a bunch of memory pool stuff in short circuit and for this api I used the name 'pre-reserve' rather than 'request' to make it clear that this is something you do *before* your process loop as part of activation. Since the API seems to be 'i will ask for the block later' type thing.